tools/c7n_gcp/c7n_gcp/actions/core.py

```diff
@@ -90,7 +90,10 @@ def invoke_api(self, client, op_name, params):
                 return e
             raise
 
-    def get_resource_params(self, m, r):
+    def get_resource_params(self, model, resource):
+        default = model.get_self_params(resource)
```

Comment:
not understanding why this is overriding what is effectively an abstract method, ie its raising not implemented

Reply:
I must say.. I dont have idea how to make that part right..

But if we add `get_self_params` then most of get methods can be created automatically..
And most action `get_resource_params` can be much much easier to implement without copy/pasting same things..

And I started adding support for `disks` and seems that I need to change a bit `get_self_params` (cause on disks normally last param is named `disk` but for `setLabels` for whatever reason its `resource`, and I see its happening for many operations)

Reply:
reverted